### PR TITLE
mev-boost: init at 1.6

### PR DESCRIPTION
**Pull request by ekimber**
## Description of changes

MEV-Boost is middleware that allows Ethereum validators to access the block-building market.

See project github [here](https://github.com/flashbots/mev-boost).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -5034,6 +5034,12 @@
     name = "Edvin Källström";
     githubId = 84442052;
   };
+  ekimber  = {
+    email = "ekimber@protonmail.com";
+    github = "ekimber";
+    name = "Edward Kimber";
+    githubId = 99987;
+  };
```
Comment (MikaelFangel) [inline maintainers/maintainer-list.nix:5042]: This should be committed by itself and appear as the first commit in this pr.

The format for the commit should then be `maintainers: add ekimber`

File: pkgs/applications/blockchains/mev-boost/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+}:
+
+buildGoModule rec {
+  pname = "mev-boost";
+  version = "1.6";
+  src = fetchFromGitHub {
+      owner = "flashbots";
+      repo = "mev-boost";
+      rev = "v${version}";
+      sha256 = "sha256-vzgX9irpI5i85bohppyL5KWQuf71SryRu1gkhWSCVKk=";
```
Comment (MikaelFangel) [inline pkgs/applications/blockchains/mev-boost/default.nix:14]:
```suggestion
      hash = "sha256-vzgX9irpI5i85bohppyL5KWQuf71SryRu1gkhWSCVKk=";
```

File: pkgs/applications/blockchains/mev-boost/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  vendorHash = "sha256-xw3xVbgKUIDXu4UQD5CGftON8E4o1u2FcrPo3n6APBE=";
+
+  meta = with lib; {
+    description = "Ethereum block-building middleware";
+    homepage = "https://github.com/flashbots/mev-boost";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ekimber ];
+    platforms = platforms.unix;
```
Comment (MikaelFangel) [inline pkgs/applications/blockchains/mev-boost/default.nix:24]: Does this have and actual binary to run as a standalone program? If so you should add the meta tag `mainProgram = <binary-name>`

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -37132,6 +37132,8 @@ with pkgs;
 
   masari = callPackage ../applications/blockchains/masari { };
 
+  mev-boost = callPackage ../applications/blockchains/mev-boost { };
```
Comment (MikaelFangel) [inline pkgs/top-level/all-packages.nix:37135]: If possible, this should be removed and the package should then be moved to the `pkgs/by-name/so/some-example/package.nix` hierarchy, as it is the preferred way of adding new packages.

**Review by MikaelFangel — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (ekimber) [reply pkgs/top-level/all-packages.nix]: Thanks for the review. I have implemented all of your suggested changes

**Comment by RaghavSood** [comment]
Result of `nixpkgs-review pr 266712` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mev-boost</li>
  </ul>
</details>

**Review by RaghavSood — APPROVED** [review]

**Pull request merged**